diff --git a/criu/cr-service.c b/criu/cr-service.c
index 61a04c5ff..c0cce482e 100644
--- a/criu/cr-service.c
+++ b/criu/cr-service.c
@@ -285,6 +285,12 @@ int exec_rpc_query_external_files(char *name, int sk)
 
 static char images_dir[PATH_MAX];
 
+bool file_exists(char *filename)
+{
+	struct stat buffer;
+	return (stat (filename, &buffer) == 0);
+}
+
 static int setup_opts_from_req(int sk, CriuOpts *req)
 {
 	struct ucred ids;
@@ -293,6 +299,7 @@ static int setup_opts_from_req(int sk, CriuOpts *req)
 	char images_dir_path[PATH_MAX];
 	char work_dir_path[PATH_MAX];
 	char status_fd[PATH_MAX];
+	char lazy_pages_socket_path[PATH_MAX];
 	bool output_changed_by_rpc_conf = false;
 	bool work_changed_by_rpc_conf = false;
 	bool imgs_changed_by_rpc_conf = false;
@@ -554,6 +561,14 @@ static int setup_opts_from_req(int sk, CriuOpts *req)
 		opts.lazy_pages = req->lazy_pages;
 	}
 
+	strcpy(lazy_pages_socket_path, images_dir);
+	strcat(lazy_pages_socket_path, "/lazy-pages.socket");
+
+	if (file_exists(lazy_pages_socket_path)) {
+		// always enable lazy-pages if the socket exists
+		opts.lazy_pages = true;
+	}
+
 	if (req->has_pre_dump_mode) {
 		switch (req->pre_dump_mode) {
 		case CRIU_PRE_DUMP_MODE__SPLICE:
