diff --git a/criu/net.c b/criu/net.c
index 2793b18e6..d0bd47fe8 100644
--- a/criu/net.c
+++ b/criu/net.c
@@ -3018,38 +3018,6 @@ int netns_keep_nsfd(void)
 	return ret >= 0 ? 0 : -1;
 }
 
-/*
- * If we want to modify iptables, we need to received the current
- * configuration, change it and load a new one into the kernel.
- * iptables can change or add only one rule.
- * iptables-restore allows to make a few changes for one iteration,
- * so it works faster.
- */
-static int iptables_restore(bool ipv6, char *buf, int size)
-{
-	int pfd[2], ret = -1;
-	char *cmd4[] = { "iptables-restore", "-w", "--noflush", NULL };
-	char *cmd6[] = { "ip6tables-restore", "-w", "--noflush", NULL };
-	char **cmd = ipv6 ? cmd6 : cmd4;
-
-	if (pipe(pfd) < 0) {
-		pr_perror("Unable to create pipe");
-		return -1;
-	}
-
-	if (write(pfd[1], buf, size) < size) {
-		pr_perror("Unable to write iptables configugration");
-		goto err;
-	}
-	close_safe(&pfd[1]);
-
-	ret = cr_system(pfd[0], -1, -1, cmd[0], cmd, 0);
-err:
-	close_safe(&pfd[1]);
-	close_safe(&pfd[0]);
-	return ret;
-}
-
 static inline int nftables_lock_network_internal(void)
 {
 #if defined(CONFIG_HAS_NFTABLES_LIB_API_0) || defined(CONFIG_HAS_NFTABLES_LIB_API_1)
@@ -3104,27 +3072,8 @@ out:
 
 static int iptables_network_lock_internal(void)
 {
-	char conf[] = "*filter\n"
-		      ":CRIU - [0:0]\n"
-		      "-I INPUT -j CRIU\n"
-		      "-I OUTPUT -j CRIU\n"
-		      "-A CRIU -m mark --mark " __stringify(SOCCR_MARK) " -j ACCEPT\n"
-									"-A CRIU -j DROP\n"
-									"COMMIT\n";
-	int ret = 0;
-
-	ret |= iptables_restore(false, conf, sizeof(conf) - 1);
-	if (kdat.ipv6)
-		ret |= iptables_restore(true, conf, sizeof(conf) - 1);
-
-	if (ret)
-		pr_err("Locking network failed: iptables-restore returned %d. "
-		       "This may be connected to disabled "
-		       "CONFIG_NETFILTER_XT_MARK kernel build config "
-		       "option.\n",
-		       ret);
-
-	return ret;
+	pr_info("skipping iptables lock network\n");
+	return 0;
 }
 
 int network_lock_internal(void)
@@ -3174,19 +3123,8 @@ static inline int nftables_network_unlock(void)
 
 static int iptables_network_unlock_internal(void)
 {
-	char conf[] = "*filter\n"
-		      ":CRIU - [0:0]\n"
-		      "-D INPUT -j CRIU\n"
-		      "-D OUTPUT -j CRIU\n"
-		      "-X CRIU\n"
-		      "COMMIT\n";
-	int ret = 0;
-
-	ret |= iptables_restore(false, conf, sizeof(conf) - 1);
-	if (kdat.ipv6)
-		ret |= iptables_restore(true, conf, sizeof(conf) - 1);
-
-	return ret;
+	pr_info("skipping iptables unlock network\n");
+	return 0;
 }
 
 static int network_unlock_internal(void)
